opendocdb-cts/internal/mongosh/fixture_test.go

```diff
@@ -162,9 +182,73 @@ func TestConvertFixtures(t *testing.T) { //nolint:revive // exceeds number of li
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
-			actual, err := ConvertFixtures(tc.fixtures)
+			t.Parallel()
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			ctx := context.Background()
+			l := slogt.New(t)
+			dbName := strings.ReplaceAll(t.Name(), "/", "-")
+
+			actualJS, err := ConvertFixtures(tc.fixtures)
+			require.NoError(t, err)
+			assert.Equal(t, unindent(tc.expected)+"\n", actualJS)
+
+			mountPath := filepath.Join(mountDir, name+".js")
+			err = os.WriteFile(mountPath, []byte(actualJS), 0o666)
+			require.NoError(t, err)
+
+			t.Cleanup(func() {
+				require.NoError(t, os.Remove(mountPath))
+			})
+
+			conn, err := wireclient.Connect(ctx, "mongodb://127.0.0.1:27001/", l)
 			require.NoError(t, err)
-			assert.Equal(t, unindent(tc.expected)+"\n", actual)
+
+			t.Cleanup(func() {
+				require.NoError(t, conn.Close())
+			})
+
+			require.NoError(t, conn.Ping(ctx))
+
+			_, _, err = conn.Request(ctx, wire.MustOpMsg(
+				"dropDatabase", int32(1),
+				"$db", dbName,
+			))
```

Comment:
`dropDatabase` seems to belong to cleanup? Unless we want to keep collection after the test.

```
		t.Cleanup(func() {
			_, _, err = conn.Request(ctx, wire.MustOpMsg(
				"dropDatabase", int32(1),
				"$db", dbName,
			))
		})
```

Reply:
I think it's always better to ensure the database is clean before running the test. In that way we're not dependent on some external factors (such as modifying the database manually from mongosh). Apart from that, even if `cleanup` runs on panics, there might be still some cases when it's impossible to recover a panic (out of memory errors for example).

About keeping the collection after the test - I've already used leftover collection to debug decimal128 test, so I think it's alright to keep them after the test.